File: Obliger/Uke 10 og 11 peasent/celle.py
class Celle:
    # Konstruktør
    def __init__(self):
        self._status = "doed"
        self._naboer = []
        self._ant_levende_naboer = 0
    
    def sett_doed(self):
        self._status = "doed"

    def sett_levende(self):
        self._status = "levende"

    def legg_til_nabo(self, nabo):
        self._naboer.append(nabo)

    def er_levende(self):
        if self._status == "levende":
            return True
        return False

    def hent_status(self):
        pass

    def hent_status_tegn(self):
        if self.er_levende():
            return "O"
        return "."

    def tell_levende_naboer(self):
        levende_naboer = []

        for celle in self._naboer:
            if celle.er_levende():
                levende_naboer.append(celle)

        self._ant_levende_naboer = len(levende_naboer)

    def oppdater_status(self):
        if self.er_levende():
            if self._ant_levende_naboer < 2:
                self.sett_doed()
            if self._ant_levende_naboer >= 2:
                self.sett_levende()
            if self._ant_levende_naboer > 3:
                self.sett_doed()
        
        if not self.er_levende():
            if self._ant_levende_naboer == 3:
                self.sett_levende()

File: IN1010/Obliger2/Oblig1/Celle.java



public class Celle {
    public Boolean levende;
    public Celle[] naboer;
    public int antNaboer;
    public int antLevendeNaboer;

    public Celle() {
        levende = false;
        naboer = new Celle[8];
        antNaboer = 0;
        antLevendeNaboer = 0;
    }

    public void settLevende() {
        levende = true;
    }

    public void settDoed() {
        levende = false;
    }

    public Boolean erLevende() {
        return levende;
    }

    public char hentStatusTegn() {
        if (erLevende()) {
            return 'O';
        } else {
            return '.';
        }
    }

    public void leggTilNabo(Celle nabo) {
        for (int i = 0; i < naboer.length; i++) {
            if (naboer[i] == null && nabo != null) {
                naboer[i] = nabo;
                antNaboer++;
                break;
            }
        }
    }

    public void tellLevendeNaboer() {
        antLevendeNaboer = 0;
        for (Celle celle : naboer) {
            if (celle != null) {
                if (celle.erLevende()) antLevendeNaboer++;
            }
        }
    }

    public void oppdaterStatus() {
        if (antLevendeNaboer < 2 || antLevendeNaboer > 3) {
            settDoed();
        } else if (this.antLevendeNaboer == 3) {
            this.settLevende();
        }
    }
}